src/quaternion_types.jl

```diff
@@ -43,10 +43,45 @@ end
 
 # These 3 functions are enough to satisfy the entire StaticArrays interface:
 function (::Type{Q})(t::NTuple{9}) where Q<:Quat
-    q = Q(sqrt(abs(1  + t[1] + t[5] + t[9])) / 2,
-               copysign(sqrt(abs(1 + t[1] - t[5] - t[9]))/2, t[6] - t[8]),
-               copysign(sqrt(abs(1 - t[1] + t[5] - t[9]))/2, t[7] - t[3]),
-               copysign(sqrt(abs(1 - t[1] - t[5] + t[9]))/2, t[2] - t[4]))
+  """
+  DERIVATION OF THE EULER–RODRIGUES FORMULA
+  FOR THREE-DIMENSIONAL ROTATIONS
+  FROM THE GENERAL FORMULA
+  FOR FOUR-DIMENSIONAL ROTATIONS
+  Johan Ernest Mebius∗
+  January 2007
+
+  https://arxiv.org/pdf/math/0701759.pdf
+  This solves the system of equations in Section 3.1
+  """
+
+  a = 1.0 + t[1] + t[5] + t[9]
+  b = 1.0 + t[1] - t[5] - t[9]
+  c = 1.0 - t[1] + t[5] - t[9]
+  d = 1.0 - t[1] - t[5] + t[9]
```

Comment:
For this to work well with any `eltype(t)`, you should avoid `Float64` literals like `1.0` and `0.5` as these will force conversion to `Float64` even if the input is `Float32`.  It's probably mostly sufficient to use an integer literal instead

```
a = 1 + t[1] + t[5] + t[9]
```

Also, with `divTerm` below, better to use

```
divTerm = 1 / (2*sqrt(a))
```

for the same reasons.